### ISSUE TITLE
Force AUG start codon in agat_sp_fix_longest_ORF.pl

### ISSUE THREAD

**Issue by nicmoya** [NBISweden/AGAT#219]
Hi,

I have a set of gene predictions with 5' partial CDS (lack a proper start codon). I used agat_sp_fix_longest_ORF.pl, but approximately half of my new protein sequences start with L. This is not realistic considering that non-AUG start codons are extremely rare in my organism. 

I was wondering if it would be possible to force agat_sp_fix_longest_ORF.pl to **only** consider alternative ORFs that start with AUG (even if it means having a shorter ORF). I know very little Perl, so I am unsure if its possible to modify the codon table to force such condition.

Please let me know.

Thanks in advance,
Nic M.

**Comment by Juke34** [comment]
Hi, I think you just need to change the codon table with the appropriate parameter. There is a strict table whitout L as possible start codon.

**Comment by Juke34** [comment]
I realised that this strict table does not exists in NCBI (https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi) but only in bioperl so, it was not necessary obvious what I was saying.
So use you must use `--ct 0`

**Comment by nicmoya** [comment]
Hi @Juke34,

I ran the script using the --ct 0 parameter. I am still seeing thousands of proteins sequences that begin with L.

Here is the report that includes the command, followed by a count of protein sequences beginning with M or L.

Am I doing something wrong? 

```
(analysis) [ndm7437@quser23 masked_merger]$ cat QX1410.stringtie.fixCDS_fM-report.txt 
usage: /home/ndm7437/.conda/envs/agat_newenv/bin/agat_sp_fix_longest_ORF.pl --gff ../../stringtie/QX1410.stringtie.TDm70.gff3 -f ../../../../genomes/QX1410/decompressed/QX1410.genome.v2.1.rf.fa --ct 0 -m 1,3,4,5 -o QX1410.stringtie.fixCDS_fM.gff
Codon table used:1
Results:
6926 genes have been modified. These genes have  8677 mRNA, and we fixed the ORF of 7731 of them.
2245 model1: Prediction contains the original prediction but is longer.
205 model3: Longer prediction found overlapping the original one but doesn't contain it(frame different).
0 model4: The prediction is shorter due to the presence of stop codon in the original CDS.
5281 model5: The prediction is shorter but the original CDS sequence has not premature stop codon
  The original CDS does not start by a start codon, it is probably incomplete or fragmented
  (The adjacent sequence of the start side might be NNNN or XXXX). 
  The prediction is probably shorter because it is forced here to use a start codon.

/!\ Remind: L and M are AA are possible start codons for standard codon table (table 1).
Particular case: If we have a triplet as WTG, AYG, RTG, RTR or ATK it will be seen as a possible start codon (but translated into X)
(analysis) [ndm7437@quser23 masked_merger]$ grep -A1 "^>" QX1410.stringtie.fixCDS_fM-all.prot.fa | grep -c "^M"
8474
(analysis) [ndm7437@quser23 masked_merger]$ grep -A1 "^>" QX1410.stringtie.fixCDS_fM-all.prot.fa | grep -c "^L"
3801
```

Thanks,
Nic

**Comment by Juke34** [comment]
You need a newer version of bioperl. The selection of table 0 was bugged in the past I had fixed that. I do not remember in which bioperl version it has been fixed.

**Comment by nicmoya** [comment]
Ah! It seems that the Bio::Tools::CodonTable bug is present in the latest bioperl version available through conda (1.7.2)

I will play with cpanm to see if I can update bioperl to 1.7.8.

**Comment by nicmoya** [comment]
I updated bioperl to 1.7.8 and the issue is resolved. Thanks for the help!
